applications/tari_base_node/src/builder.rs

```diff
@@ -230,6 +230,7 @@ impl NodeContainer {
     }
 }
 
+/// Struct to hold the base node context
```

Comment:
These comments are a start, but they don't really add any context. 

a struct called `BaseNodeContext` is obviously a struct to hold the base node context; you don't need to know _anything_ about the code to guess this much.

A good doc string includes contextual and other motivating comments, e.g.
(P.S. As I was writing this, I noticed that the struct should be private; and checked that the tests still pass -- so this is a change you could make, and a benefit of "auditing" the code by writing good docs.

```
/// The base node context is a container for all the key structural pieces for the base node application, including the 
/// communications stack, the node state machine, the miner and handles to the various services that are registered 
/// on the comms stack.
///
/// `BaseNodeContext` is not intended to be ever used directly, so is a private struct. It is only ever created in the 
// [NodeContainer] enum, which serves the purpose  of abstracting the specific `BlockchainBackend` instance away 
/// from users of the full base node stack.
```

Reply:
The same comment applies for all the single-line tautological comments here. That said, private structs and functions don't nee as much detail as public ones; just enough to explain to future devs what the function / method does (assuming it's not self-explanatory)